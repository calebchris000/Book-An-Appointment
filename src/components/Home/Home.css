.homepage {
  width: calc(100% - 14rem);
  position: absolute;
  top: 0;
  bottom: 0;
  right: 0;
  /* overflow: hidden; */
}

.hero {
  text-align: center;
  position: relative;
  height: 100vh;
  height: 100dvh;
  padding-top: 6rem;
  background-color: #fefefe;
}

.hero_header {
  font-weight: 900;
  font-size: 2rem;
  letter-spacing: 3px;
  word-spacing: 5px;
}

.hero_text {
  width: fit-content;

  color: #9a9a9a;
  font-weight: 600;
  word-spacing: 1px;
  font-size: 0.9rem;
  margin: 10px auto;
  padding-bottom: 2rem;
  border-bottom: 2px dashed #9a9a9a;
}

.vehicle-container {
  width: 80%;
  display: flex;
  margin: 0 auto;
  /* flex-wrap: wrap; */
  overflow-x: scroll;
  justify-content: center;
  gap: 5rem;
  /* border: 2px dashed; */
  transform: translateY(8rem);
}

.vehicle-container::-webkit-scrollbar {
    display: none;
  }

.vehicle-holder {
  display: flex;
  /* border: 2px dashed; */
  width: 16rem;
  flex-direction: column;
  justify-content: center;
  cursor: pointer;
}


.vehicle-circle {
  width: 10rem;
  height: 10rem;
  border-radius: 50%;
  box-sizing: border-box;
  display: flex;
  justify-content: center;
  align-items: center;
  margin: 0 auto;
  background-color: #a7bfd9;
}

.vehicle {
  width: calc(100% + 8rem);
  transition: transform 0.3s ease;
}


.vehicle:hover {
  transform: scale(1.03);
}

.vehicle-model {
  margin: 28px auto;
  text-align: center;
  width: fit-content;
  justify-self: center;
  font-weight: 700;
  padding: 18px 0;
  border-bottom: 2px dashed gray;
}

.vehicle-description {
  font-size: 0.875rem;
  line-height: 20px;
  color: rgb(177, 177, 177);
}

@media (max-width: 768px) {
  .homepage {
    width: 100%;
  }
}
