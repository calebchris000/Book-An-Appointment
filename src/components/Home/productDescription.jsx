import { useSelector, useDispatch } from 'react-redux';
import { useEffect } from 'react';
import { useParams } from 'react-router-dom';
import { fetchProductDescription } from '../../redux/productDescription/productDescriptionSlice';
import Loader from '../Loader/Loader';

export default function ProductDescription() {
  const dispatch = useDispatch();

  const { id } = useParams();

  const {productDescription, isLoading} = useSelector((store) => store.productDescription);

  useEffect(() => {
    dispatch(fetchProductDescription(id));
  }, [dispatch]);

  if (isLoading) {
    return (
      <h3><Loader/></h3>
    );}
//    if (isError) {
//     return (
//       <h3>Error Occured while fetchinng...</h3>
//     );
//   }
  return (
<>
    <div className='heading text-center'>
      <h3>
       Product Details {id}
      </h3>
      </div>

      {productDescription.map((element) => (

        <div key={element.id} className=" container-md display-flex justify-content-center">
            <div className='display-flex justify-content-center'>
        <img src = {element.image}/>
              </div>
              <div className='container-car-details'>
          <li>
            <div>Brand Name</div>
            <div>
              {element.name}
            </div>
          </li>
          <li>
            
          </li>
          <li>
            <div>Description</div>
                       <div>
              {element.description}
             
            </div>
          </li>
          <li>
            <div>Model </div>
            <div>
              {element.model}
             
            </div>
          </li>
          <li>
            <div>Engine </div>
            <div>
              {element.engine}
             
            </div>
          </li>
          <li>
            <div>Mileage</div>
            <div>
              {element.mileage}
            
            </div>
          </li>
          <li>
            <div>Price </div>
            <div>
              {element.price}
             
            </div>
          </li>
          </div>
        </div>
      ))}
    </>
  );
}
