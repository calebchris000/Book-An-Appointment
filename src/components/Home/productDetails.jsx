import { useSelector, useDispatch } from 'react-redux';
import { useEffect } from 'react';
import { useParams } from 'react-router-dom';
import { fetchProductDetails } from '../../redux/productDetails/productDetailsSlice';

export default function ProductDetails() {
  const dispatch = useDispatch();

  const { id } = useParams();

  const {productDetails, isLoading} = useSelector((store) => store.productDetails);

  useEffect(() => {
    dispatch(fetchProductDetails());
  }, [dispatch]);

  if (isLoading) {
    return (
      <h3>Loading...</h3>
    );}
//    if (isError) {
//     return (
//       <h3>Error Occured while fetchinng...</h3>
//     );
//   }
  return (
    <div>
      <h3>
        {id}
        {' '}
        Product Details
      </h3>

      {productDetails.map((element) => (

        <div key={element.id} className="stock-details">

          <li>
            <div>Symbol </div>
            <div>
              {element.name}
              {' '}
            </div>
          </li>
          <li>
            <div>Name </div>
            <div>
              {element.image}
              {' '}
            </div>
          </li>
          <li>
            {' '}
            <div>Price</div>
            {' '}
            <div>
              {element.description}
              {' '}
            </div>
          </li>
          <li>
            <div>Changes Percentage </div>
            <div>
              {element.mileage}
              {' '}
            </div>
          </li>
          <li>
            <div>Change </div>
            <div>
              {element.model}
              {' '}
            </div>
          </li>
          <li>
            <div>Volume </div>
            <div>
              {element.price}
              {' '}
            </div>
          </li>
          <li>
            <div>Day Low </div>
            <div>
              {element.engine}
              {' '}
            </div>
          </li>
          <li>
            <div> Day High </div>
            <div>
              {element.engine}
              {' '}
            </div>
          </li>
          <li>
            <div>Year Low </div>
            <div>
              {element.price}
              {' '}
            </div>
          </li>
          <li>
            <div>Year High </div>
            <div>
              {element.price}
              {' '}
            </div>
          </li>

        </div>
      ))}
    </div>
  );
}
