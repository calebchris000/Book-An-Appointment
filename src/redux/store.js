import { configureStore } from "@reduxjs/toolkit";
import productDetailsReducer from "./productDetails/productDetailsSlice";
import productDescriptionReducer from "./productDescription/productDescriptionSlice";
import thunk from "redux-thunk";
import deleteCarReducer from './deleteCarSlice';

const store = configureStore({
  reducer: {
    productDetails: productDetailsReducer,
    productDescription: productDescriptionReducer,
  },
  middleware: [thunk],

  reducer: {
    deleteCar: deleteCarReducer,
  },
});

export default store;
