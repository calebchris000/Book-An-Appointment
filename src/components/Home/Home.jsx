import React, { useEffect } from "react";
import "./Home.css";
import { useDispatch, useSelector } from "react-redux";
import { fetchProductDetails } from "../../redux/productDetails/productDetailsSlice";
import Loader from "../Loader/Loader";
import { useNavigate } from 'react-router-dom';
const Vehicle = ({ model, description, image }) => {
  return (
    <>
      <div className="vehicle-holder">
        <div className="vehicle-circle">
          <img className="vehicle" src={image} alt="" />
        </div>
        <h3 className="vehicle-model">{model.toUpperCase()}</h3>
        <p className="vehicle-description">{description}</p>
      </div>
    </>
  );
};

const Home = () => {
  const navigate = useNavigate();
  const { isLoading, Products } = useSelector((store) => store.productDetails);
  const dispatch = useDispatch();
  useEffect(() => {
    dispatch(fetchProductDetails());
  }, []);
  const name = localStorage.getItem('name')

  if (isLoading) {
    return (
      <section className="homepage">
        <Loader />
      </section>
    );
  }
  return (
    <section className="homepage">
      <div className="hero">
        <h1 className="hero_header">Welcome Back, {name}</h1>
        <p className="hero_text">Please select a Model</p>
        <div className="vehicle-container" >
          {Products.map((car) => (
            <div onClick={() => navigate(`cars/${car.id}`)}>
            <Vehicle key={car.id} model={car.model} description={car.description} image={car.image} />
            </div>
          ))}
        </div>
      </div>
    </section>
  );
};

export default Home;
