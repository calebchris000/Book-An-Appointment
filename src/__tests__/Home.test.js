import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../redux/store';
import Home from '../components/Home/Home';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <Router>
          <Home />
        </Router>
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});