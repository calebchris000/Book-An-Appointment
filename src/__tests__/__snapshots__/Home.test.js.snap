// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<section
  className="homepage"
>
  <div
    className="hero"
  >
    <h1
      className="hero_header"
    >
      Welcome Back, 
    </h1>
    <p
      className="hero_text"
    >
      Please select a Model
    </p>
    <div
      className="vehicle-container"
    />
  </div>
</section>
`;
